ibazel/ibazel_test.go

```diff
@@ -216,7 +212,7 @@ func TestIBazelLoop(t *testing.T) {
 	assertRun()
 	assertState(WAIT)
 	// Source file change.
-	i.sourceEventHandler.SourceFileEvents <- fsnotify.Event{Op: fsnotify.Write, Name: "/path/to/foo"}
+	go func () { i.sourceFileWatcher.Events() <- fs_watcher.Event{Op: fs_watcher.Write, Name: "/path/to/foo"} }()
```

Comment:
Due to removing source_event_handler, i had to change this to emit async as to not block the receiving end.

Reply:
This test was very carefully constructed to be able to run single threaded with a blocking channel. It's one of the few things I think I did right in the beginning on this project 😊 .

I would like to understand what you changed about the state machine that forced this to become async. Your comment implies you understand what is going on, could you speak a little bit more about it?

Reply:
So i'm new to go, so please bare with me and correct any misconceptions i may have. 

My understanding is that go channels will block and wait for the other side to be ready to receive an event. Before my change, we had [source event handler](https://github.com/bazelbuild/bazel-watcher/blob/master/ibazel/source_event_handler.go#L45) that was listening for events on a separate thread, and re-emitting events from that thread. The need for that entire file *appeared* negated since ibazel had evolved to watch parent directories, instead of files themselves. So after removing that separate thread, this channel send appeared to block as there was not a receiver ready on the other end. 

Does that sound like its correct? If so, should I remedy this issue another way besides an inline separate thread here?